src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,98 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import React from 'react';
+
+type PaginationProps = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const arrayOfPages = Array.from(
+    { length: Math.ceil(total / perPage) },
+    (_, i) => i + 1,
+  );
+
+  const arrayOfItems = Array.from(
+    { length: Math.min(perPage, total - (currentPage - 1) * perPage) },
+    (_, i) => {
+      return `Item ${i + 1 + (currentPage - 1) * perPage}`;
+    },
+  );
+
+  return (
+    <div>
+      <ul className="pagination">
+        <li
+          className={classNames(`page-item`, { disabled: currentPage === 1 })}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage === 1}
+            onClick={() => {
+              if (currentPage > 1) {
+                onPageChange(currentPage - 1);
+              }
+            }}
```

Comment:
It's a good practice to prevent the default action when handling `onClick` events for anchor tags to avoid any unintended page jumps or URL changes. You can do this by accepting the event parameter and calling `event.preventDefault()`.